tools/docs/rst_check.py

```diff
@@ -9,6 +9,10 @@
 RELOADABLE_FLAG_REGEX = re.compile(r".*(...)(envoy.reloadable_features.[^ ]*)\s.*")
 VERSION_HISTORY_NEW_LINE_REGEX = re.compile(r"\* ([a-z \-_]+): ([a-z:`]+)")
 VERSION_HISTORY_SECTION_NAME = re.compile(r"^[A-Z][A-Za-z ]*$")
+# Make sure backticks come in pairs.
+# Exceptions: reflinks (ref:`` where the backtick won't be preceded by a space
+#             links `title <link>`_ where the _ is checked for in the regex.
+BAD_TICKS_REGEX = re.compile(r".* `[^`].*`[^_]")
```

Comment:
im wondering will this work across lines ?

Reply:
It should - AFIK the rst script collects a line's worth of data in line and only then does checks like "does this release note have a terminal period"

It will miss things like (``header name``) so it's still best effort but IMO better than nothing.

Reply:
yep, for sure

the reason i hadnt added this previously is that a line regex wont catch situations like

```rst

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas nec porttitor tellus, commodo tincidunt elit. ``Nullam
auctor neque`` pulvinar ipsum congue, a lacinia nulla sagittis.
```

which i think is valid rst etc

my vague intention on this was to add a sphinx plugin that hooked in to the "default role" (i think that is what its called in rstspeak) and errored

but, yep, this should catch most i think